lightly/models/utils.py

```diff
@@ -474,6 +478,40 @@ def random_token_mask(
     return idx_keep, idx_mask
 
 
+def random_prefix_mask(
+    size: Tuple[int, int],
+    max_prefix_length: int,
+    device: Optional[Union[torch.device, str]] = None,
+) -> torch.Tensor:
+    """Creates a random prefix mask.
+
+    The mask is created by uniformly sampling a prefix length in [0, max_prefix_length]
+    for each sequence in the batch. All tokens with an index greater or equal to
+    the prefix length are masked.
+
+    Args:
+        size:
+            Size of the token batch for which to generate masks.
+            Should be (batch_size, sequence_length).
+        max_prefix_length:
+            Maximum length of the prefix to mask.
+        device:
+            Device on which to create the mask.
+
+    Returns:
+        A mask tensor with shape (batch_size, sequence_length) where each entry
+        is True if the token should be masked and False otherwise.
+
+    """
+    batch_size, sequence_length = size
+    arange = torch.arange(sequence_length, device=device).expand(
+        batch_size, sequence_length
+    )
+    indices = torch.randint(0, max_prefix_length, (batch_size, 1), device=device)
```

Comment:
I think this should be `indices = torch.randint(0, max_prefix_length+1, (batch_size, 1), device=device)` because the high value in `torch.randint` is not included.

Reply:
We have `max_prefix_length == num_patches - 1 == 255`. This will generate indices in `[0, 254]`. Then we set the mask with `mask = arange >= indices`, which means that all indices up-to and including `245` can be unmasked while only index 255 is always masked. I think this is what we want.

Note that we pass `num_patches - 1` here: https://github.com/lightly-ai/lightly/blob/aa3dde81c46950f43cbd3c7af8163e9dcd453c78/benchmarks/imagenet/vitb16/aim.py#L76